Do not tweak CFLAGS/CXXFLAGS for -g/-O2 if not explicitly requested
by configure arguments --*able-debug/--*able-optimize.

--- mico/configure.in
+++ mico/configure.in
@@ -112,12 +112,12 @@
  use_repo=$enableval, use_repo=no)
 
 AC_ARG_ENABLE(optimize,
- [  --disable-optimize      do not use -O2 to compile],
- use_opt=$enableval, use_opt=yes)
+ [  --enable-optimize       use -O2 to compile],
+ use_opt=$enableval, use_opt=default)
 
 AC_ARG_ENABLE(debug,
  [  --enable-debug          use -g to compile],
- use_debug=$enableval, use_debug=no)
+ use_debug=$enableval, use_debug=default)
 
 AC_ARG_ENABLE(shared,
  [  --disable-shared        don't build shared libs],
@@ -976,7 +976,7 @@
       CONF_SHARED_CC="$CXX -shared"
       CONF_PICFLAGS=-fPIC
       CONF_LDSOFLAGS=-fPIC
-      if test X"$use_opt" != Xyes; then
+      if test X"$use_opt" = Xno; then
         # by default Intel uses -O so we need to disable it
         CONF_OPT_FLAGS=-O0
         use_opt=yes
@@ -1032,17 +1032,23 @@
 #
 
 changequote(<<, >>)
-CXXFLAGS=`echo " $CXXFLAGS " | sed -e 's/ -g / /g' -e 's/ -O[0-9]* / /g'`
-CFLAGS=`echo " $CFLAGS " | sed -e 's/ -g / /g' -e 's/ -O[0-9]* / /g'`
-changequote([, ])
-if test X"$use_opt" != Xno; then
-  CXXFLAGS="$CONF_OPT_FLAGS $CXXFLAGS"
-  CFLAGS="$CONF_OPT_FLAGS $CFLAGS"
-fi
-if test X"$use_debug" != Xno; then
-  CXXFLAGS="$CONF_DEBUG_FLAGS $CXXFLAGS"
-  CFLAGS="$CONF_DEBUG_FLAGS $CFLAGS"
+if test X"$use_opt" != Xdefault; then
+  CXXFLAGS=`echo " $CXXFLAGS " | sed -e 's/ -O[0-9]* / /g'`
+  CFLAGS=`echo " $CFLAGS " | sed -e 's/ -O[0-9]* / /g'`
+  if test X"$use_opt" = Xyes; then
+    CXXFLAGS="$CONF_OPT_FLAGS $CXXFLAGS"
+    CFLAGS="$CONF_OPT_FLAGS $CFLAGS"
+  fi
 fi
+if test X"$use_debug" != Xdefault; then
+  CXXFLAGS=`echo " $CXXFLAGS " | sed -e 's/ -g / /g'`
+  CFLAGS=`echo " $CFLAGS " | sed -e 's/ -g / /g'`
+  if test X"$use_debug" = Xyes; then
+    CXXFLAGS="$CONF_DEBUG_FLAGS $CXXFLAGS"
+    CFLAGS="$CONF_DEBUG_FLAGS $CFLAGS"
+  fi
+fi
+changequote([, ])
 
 if test "x$use_memcheck" = "xyes" ; then
     case $target in
